### PR TITLE
Fix  issue 20665 - std.concurrency.spawn should document not working with delegates

### PR DESCRIPTION
**Pull request by Biotronic**
Improved compiler error message and amended documentation to document the limitation.

### CODE REVIEW

**Comment by thewilsonator** [comment]
```
std/concurrency.d(595): Error: static assert:  `__traits(compiles, spawn(F, T...)(F fn, T args) if (isSpawnable!(F, T))(dg1, 1))` is false
```

File: std/concurrency.d [hunk 1]
```diff
@@ -498,6 +501,12 @@ if (isSpawnable!(F, T))
     assert(receiveOnly!string == "This is so great!");
 }
 
+void spawn(F, T...)(F fn, T args)
```
Comment (kubo39) [inline std/concurrency.d:504]: Could you explain why use `void` instead of `Tid` ?

Comment (Biotronic) [reply std/concurrency.d:504]: Because it's going to fail anyway, I guess. I didn't really think about it. Mayhap `Tid` would be better.

**Comment by RazvanN7** [comment]
@thewilsonator I have modified this PR to reject only delegates that are not shared or immutable and documented that behavior. Since I deviated from the original fix, I think someone else shoud merge this.

**Comment by thewilsonator** [comment]
```
std/concurrency.d(520): Error: static assert:  "Cannot spawn a delegate that is not `shared` or `immutable`"
generated/linux/debug/64/publictests/std_concurrency.d(107):        instantiated from here: `spawn!(void function(Tid) @system, Tid)`
```

**Comment by RazvanN7** [comment]
Apparently, `is(void delegate(int) shared == shared)` yields false due to: https://issues.dlang.org/show_bug.cgi?id=1983 . Damn it!

**Comment by RazvanN7** [comment]
I gave up on improving the error message. I updated the documentation. Delegates that mutate unshared data are still going to fail with a non-perfect error mesage, but at least the thing is documented now.

@thewilsonator I think that we can merge this now.

**Pull request merged**